### PR TITLE
Move expression parser into java-lib

### PR DESCRIPTION
**Pull request by basilisk487**

### CODE REVIEW

File: java-lib/src/main/java/com/wavefront/predicates/EvalExpressionVisitorImpl.java [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+          toReturn *= 1E-24;
+          break;
+        default:
+          throw new IllegalArgumentException("Unknown SI Suffix: " + suffix);
+      }
+    }
+    return toReturn;
+  }
+
+  private static int murmurhash3_x86_32(CharSequence data) {
```
Comment (panghy) [inline java-lib/src/main/java/com/wavefront/predicates/EvalExpressionVisitorImpl.java:388]: Can you use the Guava hash function for this (not as efficient but probably more maintainable?)

File: java-lib/src/main/java/com/wavefront/predicates/MathExpression.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        return (long) left.getValue(entity) | (long) right.getValue(entity);
+      case "^":
+        return (long) left.getValue(entity) ^ (long) right.getValue(entity);
+      case ">>":
+        return (long) left.getValue(entity) >> (long) right.getValue(entity);
+      case ">>>":
+        return (long) left.getValue(entity) >>> (long) right.getValue(entity);
+      case "<<":
+      case "<<<":
+        return (long) left.getValue(entity) << (long) right.getValue(entity);
```
Comment (panghy) [inline java-lib/src/main/java/com/wavefront/predicates/MathExpression.java:63]: is this intentional? (merging << and <<<)

File: java-lib/src/main/java/com/wavefront/predicates/MultiStringComparisonExpression.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    switch (matchOp) {
+      case ALL:
+        return asDouble(annotations.stream().allMatch(x -> cmp.apply(x, arg.getString(entity))));
+      case ANY:
+        return asDouble(annotations.stream().anyMatch(x -> {
+          return cmp.apply(x, arg.getString(entity));
+        }));
+        //return asDouble(annotations.stream().anyMatch(x -> cmp.apply(x, arg.getString(entity))));
```
Comment (panghy) [inline java-lib/src/main/java/com/wavefront/predicates/MultiStringComparisonExpression.java:78]: remove or comment

File: java-lib/src/main/java/com/wavefront/predicates/MathExpression.java (hunk 2 above)
Comment (basilisk487) [reply java-lib/src/main/java/com/wavefront/predicates/MathExpression.java:63]: Yup, signed and unsigned left shift yield the same result

File: java-lib/src/main/java/com/wavefront/predicates/EvalExpressionVisitorImpl.java (hunk 1 above)
Comment (basilisk487) [reply java-lib/src/main/java/com/wavefront/predicates/EvalExpressionVisitorImpl.java]: Done

**Comment by basilisk487** [comment]
> Can you attach a coverage report on what the tests themselves cover in terms of the (new) classes that you're adding?

see below, +added jacoco to enforce 96% line coverage. 

image

**Pull request merged**